nifi-nar-bundles/nifi-jwt-bundle/nifi-jwt-processors/src/main/java/org/apache/nifi/processors/jwt/CustomValidators.java

```diff
@@ -0,0 +1,114 @@
+                    context);
+            if (!pathValidatorResult.isValid()) {
+                return pathValidatorResult;
+            }
+            // path is directory
+            final File directory = new File(input);
+            if (!directory.isDirectory()) {
+                return new ValidationResult.Builder().subject(subject).input(input)
+                        .explanation(subject + " must be a directory").valid(false).build();
+            }
+            // directory contains at least one ".pub" file
+            final File[] keys = directory.listFiles((dir, name) -> name.toLowerCase().endsWith(".pub"));
+            if (keys.length < 1) {
+                return new ValidationResult.Builder().subject(subject).input(input).explanation(
+                        "the directory must contain at least one RSA public key file with the entension \".pub\"")
+                        .valid(false).build();
+            }
+            return new ValidationResult.Builder().subject(subject).input(input)
+                    .explanation("Valid public key directory").valid(true).build();
+        }
+    };
+
+    public static final Validator PRIVATE_KEY_VALIDATOR = new Validator() {
+        @Override
+        public ValidationResult validate(String subject, String input, ValidationContext context) {
+            // expression language
+            if (context.isExpressionLanguageSupported(subject) && context.isExpressionLanguagePresent(input)) {
+                return new ValidationResult.Builder().subject(subject).input(input)
+                        .explanation("Expression Language Present").valid(true).build();
+            }
+            // not empty
+            ValidationResult nonEmptyValidatorResult = StandardValidators.NON_EMPTY_VALIDATOR.validate(subject, input,
+                    context);
+            if (!nonEmptyValidatorResult.isValid()) {
+                return nonEmptyValidatorResult;
+            }
+            // valid path
+            ValidationResult pathValidatorResult = StandardValidators.FILE_EXISTS_VALIDATOR.validate(subject, input,
+                    context);
+            if (!pathValidatorResult.isValid()) {
+                return pathValidatorResult;
+            }
+            // is a file
+            final File key = new File(input);
+            if (!key.isFile()) {
+                return new ValidationResult.Builder().subject(subject).input(input)
+                        .explanation(subject + " must be path to private key, not a directory").valid(false).build();
+            }
+            // file is readable
+            if (!key.canRead()) {
+                return new ValidationResult.Builder().subject(subject).input(input)
+                        .explanation(subject + " must be readable by the NiFi user").valid(false).build();
+            }
+            // read key from file
+            KeyProvider keyProvider;
+            try (final SSHClient sshClient = new SSHClient()){
```

Comment:
I now see this use of `sshj` -- is it necessary or can the private key be loaded and parsed without this dependency?

Reply:
This was the best way I could reliably and easily parse private keys created using either openssh or openssl RSA key generators

Reply:
I think we should write internal methods to perform this to avoid bringing in an unnecessary dependency. I think we already have some asymmetric key parsing utilities in `nifi-security-utils`.

Reply:
I can use bouncyCastle and standard libraries to read RSA private keys but not those generated using `ssh-keygen -t rsa ...` (which end up with headers of `-----BEGIN OPENSSH PRIVATE KEY-----` instead of `-----BEGIN RSA PRIVATE KEY-----`).

If we want to support these openssh keys it would definitely be worth writing internal methods but I am unsure where these should be, and think it would be useful to do so on a new JIRA ticket and PR...

I couldn't find anything that jumped out at me from the `nidi-security-utils` docs